### PR TITLE
Adding possibility to report expected error

### PR DESCRIPTION
**Pull request by igor-ribeiiro**
Adding functions:
```reportSuccess```
```reportExpectedException```
```reportException```

And deprecating old ones.
Adding ```expectedError``` class.

Tested by calling each function and seeing the results in the logs.

### CODE REVIEW

File: packages/office-addin-usage-data/src/usageData.ts [hunk 1]
```diff
@@ -271,10 +285,39 @@ export class OfficeAddinUsageData {
     delete this.usageDataClient.context.tags["ai.user.accountId"]; // subscription
   }
 
+   /**
+   * Transform the project name by adddin '-test' suffix to it if necessary
+   */
+  private getEventName() {
```
Comment (akrantz) [inline packages/office-addin-usage-data/src/usageData.ts:291]: Could you please move the functions so they are in alphabetic order?

**Review by akrantz — CHANGES_REQUESTED** [review]
Looks good. I think the diffs will be easier to read if the functions are in alphabetic order.

File: packages/office-addin-usage-data/src/usageData.ts (hunk 1 above)
Comment (igor-ribeiiro) [reply packages/office-addin-usage-data/src/usageData.ts:291]: I didn't even know this convention existed. 
Looking into the code, looks like the file doesn't follow this at all (I could change around the functions to fix that).
Until then, I've put it near the get function, which makes more sense.

**Review by akrantz — APPROVED** [review]
Looks good.

You could move the reportError(), reportExpectedError(), and reportSuccess() up higher so they come before the send functions and are in alphabetic order.

File: packages/office-addin-usage-data/src/usageData.ts [hunk 2]
```diff
@@ -272,44 +293,151 @@ export class OfficeAddinUsageData {
...
+  }
+
+  /**
+   * Reports custom exception event object to Application Insights
+   * @param method Method name sent to Application Insights
+   * @param err Error or message about error sent to Application Insights
+   * @param data Data object(s) sent to Application Insights
+   * @deprecated Use `reportUnexpectedError` instead.  
+   */
+  public sendUsageDataException(method: string, err: Error | string, data: object = {}) {
```
Comment (akrantz) [inline packages/office-addin-usage-data/src/usageData.ts:340]: This isn't yet in alphabetic order.

Comment (igor-ribeiiro) [reply packages/office-addin-usage-data/src/usageData.ts:340]: I have put the non-deprecated functions in alphabetic order, while the deprecated ones just after each one of them.
Do you think putting every function in alphabetic, not caring about deprecated better?

Comment (akrantz) [reply packages/office-addin-usage-data/src/usageData.ts:340]: Yes, just having them all in alphabetic order. Thanks.

**Review by akrantz — APPROVED** [review]
Looks good.

**Comment by akrantz** [comment]
@TCourtneyOwen Please take a quick look. Let me know if you have any concerns.

**Review by TCourtneyOwen — APPROVED** [review]

**Pull request merged**